stew/ranges/ptr_arith.nim

```diff
@@ -1,12 +1,17 @@
 proc baseAddr*[T](x: openarray[T]): pointer = cast[pointer](x)
 
-proc shift*(p: pointer, delta: int): pointer {.inline.} =
+# Please note that we use templates here on purpose.
+# As much as I believe in the power of optimizing compilers, it turned
+# out that the use of forced inlining with templates still creates a
+# significant difference in the release builds of nim-faststreams
+
+template shift*(p: pointer, delta: int): pointer =
   cast[pointer](cast[int](p) + delta)
 
-proc distance*(a, b: pointer): int {.inline.} =
+template distance*(a, b: pointer): int =
   cast[int](b) - cast[int](a)
 
-proc shift*[T](p: ptr T, delta: int): ptr T {.inline.} =
+template shift*[T](p: ptr T, delta: int): ptr T =
```

Comment:
shall we call this `offset` instead? that's consistent with `offsetof` which is typically used to measure field offsets and the like, and is consistent with rust naming as well (https://doc.rust-lang.org/std/primitive.pointer.html#method.offset). 

shift is something typically associated with moving bits around.

Reply:
Hrm, perhaps (although `offset` is not very clear to me either). 

The problem is that this is one of the most popular helpers in `ptr_arith` and it's used in many places.

Reply:
I think we'll benefit of standardizing these low-level op names across languages, makes it more approachable, specially since shift does have an established CS meaning.. we can `{.deprecated.}` the old name for a while while dealing with the fallout.

Reply:
there's `advance` of c++ std lib fame as well, as far as names go - https://en.cppreference.com/w/cpp/iterator/advance - but I think `offset` is more clear because advance in the negative direction is a bit iffy

Reply:
also, `stew` is explicitly unstable - it exists to work out the right solution and balance - in this case for naming. thanks to the git submodule stuff we can control the upgrade - a mass rename is cheap and easy to do if you think about it, it's just a mechanical change. 

I've thought of a few other things for ptr_arith to make it more complete and orthogonal, we can address it then as well.

Reply:
I always use [offset for pointer arithmetics](https://github.com/numforge/laser/blob/e797ca61f3f6a2dfc9f55ba57d14059e6278319f/laser/primitives/matrix_multiplication/gemm_utils.nim#L16-L18).